novawallet/Modules/Ramp/Hook/Transak/OffRamp/TransakOffRampResponseHandler.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+
+final class TransakOffRampResponseHandler {
+    weak var delegate: OffRampHookDelegate?
+
+    let logger: LoggerProtocol
+    let chainAsset: ChainAsset
+
+    init(
+        delegate: OffRampHookDelegate,
+        chainAsset: ChainAsset,
+        logger: LoggerProtocol
+    ) {
+        self.delegate = delegate
+        self.chainAsset = chainAsset
+        self.logger = logger
+    }
+}
+
+extension TransakOffRampResponseHandler: OffRampMessageHandling {
+    func canHandleMessageOf(name: String) -> Bool {
+        name == TransakRampEventNames.webViewEventsName.rawValue
+    }
+
+    func handle(message: Any, of _: String) {
+        guard
+            let messageData = "\(message)".data(using: .utf8)
+        else {
+            logger.error("Unexpected message: \(message)")
+            return
+        }
+
+        if
+            let transferEvent = try? JSONDecoder().decode(
+                TransakEvent<TransakTransferEventData>.self,
+                from: messageData
+            ),
+            let transferData = transferEvent.data,
+            transferEvent.eventId == .orderCreated {
+            switch transferData.status {
+            case .awaitingPayment:
+                let model = PayCardTopupModel(
+                    chainAsset: chainAsset,
+                    amount: transferData.cryptoAmount,
+                    recipientAddress: transferData.cryptoPaymentData.paymentAddress
+                )
+
+                delegate?.didRequestTransfer(from: model)
+            }
+        } else if
+            let transferEvent = try? JSONDecoder().decode(
```

Comment:
What to you think about moving decoding logic to the TransakEvent itselt. As i understand event id field is always there and data field depends on eventId. So we could make TransakEvent as enum and implement custom decoding that decodes eventId first and then decodes data based on event id. Finally, once we associate decoded data with the enum case